Libraries/Image/RCTImageUtils.m

```diff
@@ -110,6 +111,26 @@ CGRect RCTTargetRect(CGSize sourceSize, CGSize destSize,
           RCTCeilSize(sourceSize, destScale)
         };
       }
+
+    case RCTResizeModeCenter:
+
+      // Make sure the image is not clipped by the target.
+      if (sourceSize.height > destSize.height) {
+        sourceSize.width = destSize.width = destSize.width;
```

Comment:
Could be a bug in the other code then too. Could you clean up all of them up? It probably should just be `sourceSize.width = destSize.width`